### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by Ahmed-Benj**
Hello,

please reveiw my rspec module

Kind regards

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
+require_relative '../enumerable_methods'
+#rubocop:disable all
```
Comment (carlos-ssh) [inline spec/enumerable_spec.rb]: **Kindly enable Rubocop.**

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    end
+  end
+
+  describe '#my_none' do
+    it 'Returns true if no friend length = 3' do
+      expect(friends_strings.my_none { |friend| friend.length == 3 }).to eq(friends_strings.none? do |friend|
+                                                                              friend.length == 3
+                                                                            end)
+    end
+  end
```
Comment (carlos-ssh) [inline spec/enumerable_spec.rb]: **Kindly provide another kind of examples:**
- All the tests that you created are passing the same thing that you give:

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hello Team 👋🏽
Your project is really nice! 

Few changes are needed:
- Add your .idea directory to your .gitignore
- Kindly provide Negative Scenarios for your Tests
Test Image 4
- Kindly work on your created Tests: (Lines of code below).
     

_Happy coding!_
@carlos-ssh

**Comment by Ahmed-Benj** [comment]
I have made the requested changes.

However I don't know why the checks don't run correctly in github : 

 don't know how to fix this

```
spec/enumerable_spec.rb:10:69: W: Lint/Void: Operator * used in void context.
      expect(ints.my_each { |x| (x * 5) }).to eq(ints.each { |x| (x * 5) })
```

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+describe Enumerable do
+  let(:ints) { [3, 4, 5] }
+  let(:friends) { %w[Sharon Leo Leila Brian Arun] }
+  let(:friends_strings) { %w[Sharon Leo Leila Brian Arun] }
+
+  describe '#my_each' do
+    it 'Executes a block of code' do
+      expect(ints.my_each { |x| (x * 5) }).to eq(ints.each { |x| (x * 5) })
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: - [x] A bit of common sense: check if specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) Example: there is a spec for correct usage of the method \[e.g. `it "creates object if all required attributes are passed"`\] and also there is spec for the wrong usage of a method \[e.g. `it "does not create the object if required attributes are missing"`\].
Kindly add negative test case scenarios to this method.

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_each_with_index' do
+    it 'Executes a block of code' do
+      expect(friends.my_each_with_index do |friend, index|
+               puts friend + index.to_s
+             end).to eq(friends.each_with_index do |friend, index|
+                          puts friend + index.to_s
+                        end)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: Also, add negative test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_all' do
+    it 'Returns true if all friends length > 5' do
+      expect(friends_strings.my_all { |friend| friend.length > 5 }).to eq(friends_strings.all? do |friend|
+                                                                            friend.length > 5
+                                                                          end)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: Kindly add negative test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                                                                              friend.length == 10
+                                                                            end)
+    end
+  end
+
+  describe '#my_none' do
+    it 'Returns true if no friend length = 3' do
+      expect(friends_strings.my_none { |friend| friend.length == 3 }).to eq(false)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: Kindly add positive test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_count' do
+    it 'Returns the number of elements in the array' do
+      expect(friends_strings.my_count).to eq(friends_strings.count)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: Kindly add negative test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_map' do
+    it 'Returns a modified array' do
+      expect(ints.my_map { |i| i * i }).to eq(ints.map { |i| i * i })
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: - [x] Kindly also add negative test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_inject' do
+    it 'Returns the sum of array elements' do
+      expect(ints.my_inject { |sum, friend| sum + friend }).to eq(ints.inject { |sum, friend| sum + friend })
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: - [x] Kindly also add negative test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_map_proc' do
+    it 'Returns a modified array using proc' do
+      proc = proc { |friend| "#{friend}s" }
+      expect(friends_strings.my_map_proc(proc)).to eq(%w[Sharons Leos Leilas Brians Aruns])
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: - [x] Kindly also add negative test case scenarios to this method. The reason, already given in the first highlight.

File: spec/enumerable_spec.rb [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe '#my_map_proc_block' do
+    it 'Returns a modified array using proc instead of block' do
+      proc = proc { |friend| friend.length > 5 ? friend : nil }
+      expect(friends_strings.my_map_proc_block(proc) { |friend| "#{friend}s" }).to eq(['Sharon', nil, nil, nil, nil])
+    end
+  end
+end
```
Comment (enaburekhan) [inline spec/enumerable_spec.rb]: - [x] Kindly also add negative test case scenarios to this method. The reason, already given in the first highlight.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi Bouhmid!
Great job so far:clap::clap:. However, we just need to make a few changes to get the project approved:muscle:.**

**To improve**

- [x] Check my inline comments and make the required changes.
- [x] Fix rubocop faiilures

Make the above-required changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

File: spec/enumerable_spec.rb (hunk 3 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: Done

File: spec/enumerable_spec.rb (hunk 4 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 5 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 6 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 7 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 8 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 9 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 10 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

File: spec/enumerable_spec.rb (hunk 11 above)
Comment (Ahmed-Benj) [reply spec/enumerable_spec.rb]: done

**Review by george-shammar — APPROVED** [review]
### **STATUS:APPROVED** :heavy_check_mark: :heavy_check_mark: :+1: 

Good job and thank you for implementing all the required changes from the previous reviews.

[OPTIONAL]

- [ ] Always follow the correct gitflow. Here, you're attempting to merge rspec directly to the main branch.
- [ ] Please change the description you have for the `it` statements for the negative test cases. The statements you have for both positive and negative tests are the same. 

I wish you goodluck in your future projects.

Cheers :smile:

**Pull request merged**